.github/workflows/unit.yaml

```diff
@@ -1,37 +1,27 @@
 name: Unit Tests
 
-on:
-  workflow_call:
-    inputs:
-      ref:
-        type: string
-      skipRefCheck:
-        type: boolean
-        default: true
+on: ["pull_request"]
```

Comment:
we should scope this down to main only since that's what we care about.

  push:
    branches: ["master"]
  pull_request:
    branches: ["master"]

also need the push: master since that's in the coveralls spec

Reply:
since we are changing this from a reusable workflow we need to remove this workflow from the other places it's called. 

https://github.com/Azure/aks-app-routing-operator/blob/main/.github/workflows/pr-validate-fork.yaml
https://github.com/Azure/aks-app-routing-operator/blob/main/.github/workflows/pr-validate.yaml

Reply:
updated, thanks for all comments and suggestions.